web/src/app/admin/admin-message-logs/AdminDebugMessagesLayout.tsx

```diff
@@ -77,50 +62,84 @@ export default function AdminDebugMessagesLayout(): JSX.Element {
     variables: { first: MAX_QUERY_ITEMS_COUNT },
   })
 
-  const [params, setParams] = useURLParams({
+  const [params] = useURLParams({
     search: '',
     start: '',
     end: '',
   })
 
-  const results = useFuse<DebugMessage>({
-    data:
-      data?.debugMessages.map((d: DebugMessage) => ({
-        ...d,
-        additionalKeys: {
-          filteredDestination: d.destination.replace('-', ''),
-        },
-      })) || [],
-    keys: [
-      'destination',
-      'userName',
-      'serviceName',
-      'status',
-      'additionalKeys.filteredDestination',
-    ],
-    search: params.search,
-    options: {
-      shouldSort: false,
-      showResultsWhenNoSearchTerm: true,
-      ignoreLocation: true,
-      useExtendedSearch: true,
-    },
-  })
+  if (error) return <GenericError error={error.message} />
+  if (loading && !data) return <Spinner />
 
   const startDT = params.start ? DateTime.fromISO(params.start) : null
   const endDT = params.end ? DateTime.fromISO(params.end) : null
 
-  const filteredResults = results.filter((result) => {
-    const createdAtDT = DateTime.fromISO(result.item.createdAt)
-    if (startDT && startDT > createdAtDT) return false
-    if (endDT && endDT < createdAtDT) return false
-    return true
-  })
+  const filteredData: DebugMessage[] = data?.debugMessages
+    .filter((msg: DebugMessage) => {
+      const createdAtDT = DateTime.fromISO(msg.createdAt)
+      if (params.search) {
+        if (
+          params.search === msg.alertID?.toString() ||
+          params.search === msg.createdAt ||
+          params.search === msg.destination ||
+          params.search === msg.serviceID ||
+          params.search === msg.serviceName ||
+          params.search === msg.userID ||
+          params.search === msg.userName
+        ) {
+          return true
+        }
+        return false
+      }
+      if (startDT && startDT > createdAtDT) return false
+      if (endDT && endDT < createdAtDT) return false
+      return true
+    })
+    .sort((_a: DebugMessage, _b: DebugMessage) => {
+      const a = DateTime.fromISO(_a.createdAt)
+      const b = DateTime.fromISO(_b.createdAt)
+      if (a < b) return 1
+      if (a > b) return -1
+      return 0
+    })
+  const paginatedData = filteredData.slice(0, numRendered)
 
-  const displayedResults = filteredResults.slice(0, numRendered)
+  const hasData = filteredData?.length > 0
+  const s = hasData
+    ? startDT ||
+      DateTime.fromISO(filteredData[filteredData.length - 1].createdAt).startOf(
+        'day',
+      )
+    : null
+  const e = hasData
+    ? endDT || DateTime.fromISO(filteredData[0].createdAt).endOf('day')
+    : null
+  let ivl: Interval | null = null
+  if (s && e && hasData) {
+    ivl = Interval.fromDateTimes(s, e)
+  }
 
-  if (error) return <GenericError error={error.message} />
-  if (loading && !data) return <Spinner />
+  const intervalType = 'daily'
+  const graphData = ivl
+    ? ivl.splitBy({ days: 1 }).map((i) => {
```

Comment:
Can we add a field to make this configurable? Daily won't be super helpful for determining if throttling is happening; we'll want hour-by-hour and even minute-by-minute

I want to be careful of scope, but I think that field is necessary to get what we need from the graph.

The alert metrics code has an example of the separate hook, bucketing, and worker code -- let me know you'd like additional help/context with this as it's relatively new in the repo